homeassistant/components/nexia/climate.py

```diff
@@ -224,20 +234,43 @@ def preset_mode(self):
         return self._zone.get_preset()
 
     async def async_set_humidity(self, humidity: int) -> None:
-        """Dehumidify target."""
-        if self._thermostat.has_dehumidify_support():
-            await self.async_set_dehumidify_setpoint(humidity)
+        """Set Dehumidify target."""
+
+        if self._zone.get_current_mode() == OPERATION_MODE_HEAT:
+            if self._thermostat.has_humidify_support():
+                await self.async_set_humidify_setpoint(humidity)
+        elif self._zone.get_current_mode() == OPERATION_MODE_COOL:
+            if self._thermostat.has_dehumidify_support():
+                await self.async_set_dehumidify_setpoint(humidity)
         else:
-            await self.async_set_humidify_setpoint(humidity)
+            if self._thermostat.has_humidify_support():
+                await self.async_set_humidify_setpoint(humidity)
+            if self._thermostat.has_dehumidify_support():
+                await self.async_set_humidify_setpoint(humidity)
```

Comment:
```suggestion
        elif self._thermostat.has_humidify_support():
            await self.async_set_humidify_setpoint(humidity)
        elif self._thermostat.has_dehumidify_support():
            await self.async_set_dehumidify_setpoint(humidity)
```

alternate suggestion

Reply:
It was a typo. Good catch. However, it's a little clunkly in that it results in a flakey UI response. Setting the value to 55%, e.g., results in an error that it's out of range for heating humidify, however the value is still clamped and set on the second adjustment. I will look into this and see if that exception can be removed now that we clamp in the module.